### DISCUSSION TITLE
Cant setup an selfhosted instance

### DISCUSSION THREAD

**Discussion by gitrys** [lukevella/rallly#200] [Q&A]
Hello there, im struggling to setup rallly on my homeserver. I'm fairly new to selfhosting and want to learn in the process. 
My problem is as follows:
If i setup rallly on my homeserver with  >NEXT_PUBLIC_BASE_URL=PrivateIP:Port< i can use it in my homenet. Normally i now just setup a reverse proxy (ngninx proxy manager) to expose the program to the internet, so that my friends can use the rallly instance. 
I setup my domain name and forward to >privateIP:Port<.  In other selfhosted projects i can now request a new SSL certificate. 
That process won't work in rallly. 

I read in other threads that:
- I have to change >NEXT_PUBLIC_BASE_URL=MyActualPublicURL(without a port i assume?)< Is that so far correct? I must not use my private IP in that env?
- I have to pass a "header with the domain from which the request originated" within the reverse proxy? To my understanding that is my public IP (sorry, no native speaker...)? So that domain could be rallly.mywebsite.com?

im really stuck here :(
Best regards

**Comment by lukevella** [comment]
`NEXT_PUBLIC_BASE_URL`  is the base url for all urls created by the client and server. If using a reverse proxy it should be set to whatever domain the reverse proxy is configured with (i.e. the public url like rallly.mywebsite.com).

If you set it to localhost:port for example but then all urls created by the client (ex. api requests) and server (ex. links in emails) will point to localhost which is probably not what you want.

That said I do plan on pushing an update soon that will make make configuring this parameter optional which should make this easier.

**Comment by joho1968** [comment]
> Hello there, im struggling to setup rallly on my homeserver. I'm fairly new to selfhosting and want to learn in the process.

[..]

> im really stuck here :( Best regards

Did you work out how to get this working? I struggled too initially, but have it working properly now with Docker and using Nginx as a reverse proxy.

**Reply by gitrys** [reply]
actually no, im struggeling as now
